### PR TITLE
5271 Translate HL7 full-ELR ORC segment to FHIR

### LINKED ISSUE #5271: Convert full ELR HL7 ORC segment data to FHIR resource for the first observation
### User Story:
As a sender, I want my all of my submitted data to be converted to FHIR, so it can be fully and accurately reported to STLTs.

### Description/Use Case
We have an initial HL7 to FHIR conversion, but we now need to verify that all data fields are correctly converted and contained in a FHIR resource to make sure we do not lose any data in the conversion.  This work is to verify that ALL data is retained without any loss for a given HL7 portion (see acceptance criteria).

### Risks/Impacts/Considerations
* May need some research as to what resources will contain a piece of data.  
* Is there a possibility we are not able to decode some HL7 data due to the current libraries we are using?
* There seem to be duplicate resources in the generated FHIR bundle, so care must be taken to remove those duplicates as needed.
* There are duplicate segments in the HCA HL7 messages that need to be handled accordingly.  May need to communicate with HCA to ask about those segments, but it is expected those are coming from the source and we will probably have to handle them.

### Dev Notes:
* The HL7 to FHIR conversion is configuration driven.
* HCA sample data is stored in Keybase.  It has no PII.
* Note there are repeating OBR segments, and each segment has its own set of segments up to the next OBR segment.

### Acceptance Criteria
* The full ELR HL7 ORC segment data is converted to FHIR resources.
* The segment's HL7 conversion is documented using the HL7 v2 to FHIR mapping information.
* All available data from the ORC segment fields of an HCA full ELR HL7 message are stored as part of the FHIR bundle.

### PR DESCRIPTION
**Pull request by carlosfelix2**
This PR translates HL7 full-ELR ORC segment to FHIR.
NOTE there were a few deviations from the HL7 mapping in the Provenance resource as noted in the spreadsheet as `Will not implement` due to the source of the data being in repeating ORC segments and the Provenance resource is not repeating.  The library does not support this type of conversion and we would not know which ORC segment the data belongs to.  I added one extra piece of data to a custom extension to the ServiceRequest resource to make sure we had all the data.

Test Steps:
1. Download the sample_hl7.txt sample full-ELR HL7 file to be used as input. This file is in a comment under #5271. Rename this file to sample_hl7.hl7.
2. Checkout this branch and build it with `gradle package`
3. Run the FHIR CLI command by running the following, replacing with the location and file name of the test file from step 1. Note must have an .hl7 extension.
`./prime fhirdata --input-file <input file> --output-file build/sample.fhir`
4. Open the file build/sample.fhir file with a text editor and inspect the data for the Patient resource
5. Open the HL7 to FHIR mapping spreadsheet attached to a comment in the #5271 issue and review the mappings.
  a. Under the DONE column, GREEN means done, RED not done. The two not done is because we do not think the data will be needed.
  b. Comment column may have the location data was added or some other comment.
  c. The rest of the table is as is from HL7.

## Changes
- Translate full-ELR HL7 ORC segment to FHIR
- Updated integration test data

## Checklist

### Testing
- [ ] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #5271

## To Be Done

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?


## Pull reviewers stats
Stats for the last 30 days:
|                                                                                                                                                                            | User               | Total reviews | Time to review                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       | Total comments |
| -------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------ | ------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | -------------- |
| <a href="https://github.com/jeremy-page"><img src="https://avatars.githubusercontent.com/u/54326889?u=600f8a60859e9b0ca4634227f00b2cd9bee5e9b8&v=4" width="20"></a>        | jeremy-page        | **27**        | [35m](https://app.flowwer.dev/charts/review-time/~(u~(i~'54326889~n~'jeremy-page)~p~30~r~(~(d~'rbh8wb~t~'sy)~(d~'rbhbdb~t~'70)~(d~'rbryjn~t~'1hry)~(d~'rbrykf~t~'1hsq)~(d~'rbrylr~t~'1hu2)~(d~'rbryn3~t~'5pnk)~(d~'rbryno~t~'5po5)~(d~'rbse3c~t~'54)~(d~'rbubyv~t~'3bo)~(d~'rc1r6v~t~'r2)~(d~'rbse2z~t~'4v)~(d~'rbse3p~t~'5d)~(d~'rbo8yc~t~'11r)~(d~'rc4uds~t~'i)~(d~'rc501s~t~'1ny)~(d~'rc4uau~t~'1mc)~(d~'rc4ucl~t~'h)~(d~'rcicp0~t~'v9)~(d~'rcht6i~t~'1fuq)~(d~'rcr64u~t~'9c)~(d~'rcek78~t~'4vo)~(d~'rcc7ng~t~'59wr)~(d~'rccf34~t~'58e1)~(d~'rcjsm1~t~'3ux)~(d~'rcrcml~t~'786y)~(d~'rcrma8~t~'1q)~(d~'rcuqf6~t~'51)~(d~'rcuq8n~t~'63)~(d~'rcuqhq~t~'idxj)~(d~'rcuphg~t~'1l0a))))  | 3              |
| <a href="https://github.com/kevinhaube"><img src="https://avatars.githubusercontent.com/u/4022304?u=c0f80e2ad03386621394d1830dc33b7b5cc23680&v=4" width="20"></a>          | kevinhaube         | 13            | [1h 53m](https://app.flowwer.dev/charts/review-time/~(u~(i~'4022304~n~'kevinhaube)~p~30~r~(~(d~'rbd28d~t~'3ps)~(d~'rbd874~t~'1lbb)~(d~'rbdbbd~t~'1ofk)~(d~'rbdbc0~t~'1h3)~(d~'rbd8a1~t~'1hfz)~(d~'rbd8aj~t~'1hgh)~(d~'rbqbw3~t~'2c)~(d~'rbq6k1~t~'1ynv)~(d~'rbq9cw~t~'jv)~(d~'rbq9fl~t~'mk)~(d~'rbgx2h~t~'1bhu)~(d~'rbh436~t~'1iij)~(d~'rbhuv7~t~'29ak)~(d~'rbmq43~t~'14i)~(d~'rbmq4a~t~'14p)~(d~'rcc92y~t~'fi)~(d~'rcg6bm~t~'7t)~(d~'rcet3i~t~'6qs)~(d~'rc6wfc~t~'2j)~(d~'rcs3pb~t~'lta)~(d~'rct01z~t~'1i5y)~(d~'rct02w~t~'1i6v)~(d~'rcifmu~t~'vn)~(d~'rct9ap~t~'3mh))))                                                                                                            | 17             |
| <a href="https://github.com/carlosfelix2"><img src="https://avatars.githubusercontent.com/u/63804190?u=009d26bf9914817d2f9a066811541d0f6f3155c9&v=4" width="20"></a>       | carlosfelix2       | 13            | [14h 15m](https://app.flowwer.dev/charts/review-time/~(u~(i~'63804190~n~'carlosfelix2)~p~30~r~(~(d~'rc1ekb~t~'u)~(d~'rc1asl~t~'1t)~(d~'rbs3qd~t~'1cb7)~(d~'rbsbos~t~'t)~(d~'rboezr~t~'7)~(d~'rbzk1a~t~'630)~(d~'rc1fvp~t~'1vu6)~(d~'rci1nv~t~'1b)~(d~'rcr0os~t~'1p63)~(d~'rcr4z1~t~'13m)~(d~'rc31gq~t~'1dl8)~(d~'rc4u0n~t~'3655)~(d~'rc4u12~t~'19z1)~(d~'rc5jdl~t~'1v8)~(d~'rc6qbz~t~'13iz)~(d~'rc6qdg~t~'13kg)~(d~'rc6qdv~t~'13kv)~(d~'rc6qe9~t~'13l9)~(d~'rc6qej~t~'13lj)~(d~'rc6qgg~t~'13ng)~(d~'rbzcqr~t~'1jl)~(d~'rchxnw~t~'1bgl)~(d~'rchxzf~t~'1bs4)~(d~'rchy5i~t~'1by7)~(d~'rchy7n~t~'1c0c)~(d~'rchyad~t~'1c32)~(d~'rchyd4~t~'1c5t)~(d~'rct8ty~t~'35q)~(d~'rbgvkj~t~'1d1l)))) | **56**         |
| <a href="https://github.com/clediggins-usds"><img src="https://avatars.githubusercontent.com/u/89804087?u=5b03415b2bc7766f96908dd40f522633187e4b02&v=4" width="20"></a>    | clediggins-usds    | 8             | [1h 3m](https://app.flowwer.dev/charts/review-time/~(u~(i~'89804087~n~'clediggins-usds)~p~30~r~(~(d~'rbqkv9~t~'2ll)~(d~'rbqlix~t~'399)~(d~'rc6ulc~t~'10f)~(d~'rccotv~t~'df)~(d~'rc746w~t~'ckd)~(d~'rc6tgh~t~'ds)~(d~'rbzlqr~t~'amv)~(d~'rcrqor~t~'lig)~(d~'rc3fwk~t~'1sp7)~(d~'rc75l3~t~'lu)~(d~'rctavi~t~'mhud)~(d~'rctb6v~t~'ir))))                                                                                                                                                                                                                                                                                                                                                | 16             |
| <a href="https://github.com/TomNUSDS"><img src="https://avatars.githubusercontent.com/u/74203452?u=6eccca154e9d5c808c5269d9604648aec3c9a412&v=4" width="20"></a>           | TomNUSDS           | 6             | [1h 19m](https://app.flowwer.dev/charts/review-time/~(u~(i~'74203452~n~'TomNUSDS)~p~30~r~(~(d~'rbdfi8~t~'1smf)~(d~'rbqff9~t~'6g)~(d~'rc36ga~t~'3n8)~(d~'rc36n7~t~'1kn5)~(d~'rbfoeo~t~'2u1)~(d~'rbh75z~t~'1llc)~(d~'rbmga9~t~'6upm)~(d~'rc73tt~t~'31)~(d~'rcfzf6~t~'27))))                                                                                                                                                                                                                                                                                                                                                                                                            | 12             |
| <a href="https://github.com/bgantick"><img src="https://avatars.githubusercontent.com/u/640182?u=901de0e5fa2ff06da055ee0949aa806a6c82dcb8&v=4" width="20"></a>             | bgantick           | 5             | [1h 22m](https://app.flowwer.dev/charts/review-time/~(u~(i~'640182~n~'bgantick)~p~30~r~(~(d~'rbd6vp~t~'8d4)~(d~'rbz79u~t~'52c2)~(d~'rbzc3r~t~'ia)~(d~'rbzcy9~t~'h1)~(d~'rc2z4i~t~'3i)~(d~'rc4srj~t~'1gyh)~(d~'rccjyn~t~'3s2))))                                                                                                                                                                                                                                                                                                                                                                                                                                                      | 1              |
| <a href="https://github.com/brick-green"><img src="https://avatars.githubusercontent.com/u/86254221?u=894ef5a4773dd01e92be595af8f1879041f85002&v=4" width="20"></a>        | brick-green        | 5             | [30m](https://app.flowwer.dev/charts/review-time/~(u~(i~'86254221~n~'brick-green)~p~30~r~(~(d~'rbf7d7~t~'sa)~(d~'rbzbjk~t~'9r)~(d~'rc55bf~t~'1eh)~(d~'rceg1r~t~'1y3)~(d~'rcctsn~t~'6l5))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           | 0              |
| <a href="https://github.com/ahay-agile6"><img src="https://avatars.githubusercontent.com/u/19178435?u=aa7d4ddd31665f83e677e20967502557741345e4&v=4" width="20"></a>        | ahay-agile6        | 4             | [2h 33m](https://app.flowwer.dev/charts/review-time/~(u~(i~'19178435~n~'ahay-agile6)~p~30~r~(~(d~'rbg05i~t~'ekv)~(d~'rbg05v~t~'el8)~(d~'rc73t6~t~'2e)~(d~'rcrdi5~t~'72h)~(d~'rck4y1~t~'57))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        | 2              |
| <a href="https://github.com/mreifman"><img src="https://avatars.githubusercontent.com/u/2730338?u=73b7df0da42b98f7eaec3fad11d501f35e9ef05a&v=4" width="20"></a>            | mreifman           | 3             | [32m](https://app.flowwer.dev/charts/review-time/~(u~(i~'2730338~n~'mreifman)~p~30~r~(~(d~'rbto84~t~'17bn)~(d~'rc73td~t~'2l)~(d~'rc5ble~t~'1hj))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   | 1              |
| <a href="https://github.com/oslynn"><img src="https://avatars.githubusercontent.com/u/20068283?u=bcad2a3d00ffe648463e3fbe1894762aeab72404&v=4" width="20"></a>             | oslynn             | 3             | [13h 24m](https://app.flowwer.dev/charts/review-time/~(u~(i~'20068283~n~'oslynn)~p~30~r~(~(d~'rchp2w~t~'117k)~(d~'rccu86~t~'ng)~(d~'rc58jg~t~'5ex1))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               | 2              |
| <a href="https://github.com/jorg3lopez"><img src="https://avatars.githubusercontent.com/u/49923512?v=4" width="20"></a>                                                    | jorg3lopez         | 3             | [2h 30m](https://app.flowwer.dev/charts/review-time/~(u~(i~'49923512~n~'jorg3lopez)~p~30~r~(~(d~'rc4y4f~t~'54i0)~(d~'rc50yb~t~'6xn)~(d~'rcjvov~t~'6xr))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            | 0              |
| <a href="https://github.com/MauriceReeves-usds"><img src="https://avatars.githubusercontent.com/u/74194189?u=f81b3f069b28469e1fb39322ba3c17112c1dd1d0&v=4" width="20"></a> | MauriceReeves-usds | 3             | [1h 50m](https://app.flowwer.dev/charts/review-time/~(u~(i~'74194189~n~'MauriceReeves-usds)~p~30~r~(~(d~'rcr50d~t~'14y)~(d~'rcctyo~t~'52w)~(d~'rcuqdp~t~'11jz))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | 0              |
| <a href="https://github.com/sethdarragile6"><img src="https://avatars.githubusercontent.com/u/92405130?u=b36435069928f7e5f03109c9f478cdc7158fc01e&v=4" width="20"></a>     | sethdarragile6     | 3             | [1h 50m](https://app.flowwer.dev/charts/review-time/~(u~(i~'92405130~n~'sethdarragile6)~p~30~r~(~(d~'rbqh6q~t~'1xx)~(d~'rbn1hp~t~'jwn)~(d~'rc6w8u~t~'4ya)~(d~'rc6wkf~t~'59v))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      | 2              |
| <a href="https://github.com/TurkeyFried"><img src="https://avatars.githubusercontent.com/u/2624105?u=675395bdf4c1b121eb2f291338079089107daf74&v=4" width="20"></a>         | TurkeyFried        | 1             | [27m](https://app.flowwer.dev/charts/review-time/~(u~(i~'2624105~n~'TurkeyFried)~p~30~r~(~(d~'rbd338~t~'17j)~(d~'rbd351~t~'19c)~(d~'rbd423~t~'26e))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                | 1              |
| <a href="https://github.com/JosiahSiegel"><img src="https://avatars.githubusercontent.com/u/5522990?v=4" width="20"></a>                                                   | JosiahSiegel       | 1             | [**3m**](https://app.flowwer.dev/charts/review-time/~(u~(i~'5522990~n~'JosiahSiegel)~p~30~r~(~(d~'rcic8f~t~'54))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   | 0              |
| <a href="https://github.com/RickHawesUSDS"><img src="https://avatars.githubusercontent.com/u/48692522?u=323d76e6f867d8ed0b3e56d7e5808313bfcc2ccf&v=4" width="20"></a>      | RickHawesUSDS      | 1             | [6m](https://app.flowwer.dev/charts/review-time/~(u~(i~'48692522~n~'RickHawesUSDS)~p~30~r~(~(d~'rccu9x~t~'9j))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     | 2              |
| <a href="https://github.com/jimduff-usds"><img src="https://avatars.githubusercontent.com/u/62258514?u=b7ce81d1478c472fb5bf6f3d31edb1a4454d55dd&v=4" width="20"></a>       | jimduff-usds       | 1             | [12h 26m](https://app.flowwer.dev/charts/review-time/~(u~(i~'62258514~n~'jimduff-usds)~p~30~r~(~(d~'rcundc~t~'yjm))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                | 0              |
| <a href="https://github.com/cwinters-usds"><img src="https://avatars.githubusercontent.com/u/86251310?u=e7f46bd48ca250dea43f9e0d6433a7fb8627ed30&v=4" width="20"></a>      | cwinters-usds      | 1             | [4h 37m](https://app.flowwer.dev/charts/review-time/~(u~(i~'86251310~n~'cwinters-usds)~p~30~r~(~(d~'rbbmge~t~'cmt)~(d~'rbbmic~t~'cor)~(d~'rbbmmo~t~'ct3)~(d~'rbbnbd~t~'dhs)~(d~'rbbnx9~t~'e3o))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | 4              |
| <a href="https://github.com/snesm"><img src="https://avatars.githubusercontent.com/u/94193373?v=4" width="20"></a>                                                         | snesm              | 1             | [10h 44m](https://app.flowwer.dev/charts/review-time/~(u~(i~'94193373~n~'snesm)~p~30~r~(~(d~'rbcykq~t~'tt4))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       | 0              |
| <a href="https://github.com/jh765"><img src="https://avatars.githubusercontent.com/u/96186164?v=4" width="20"></a>                                                         | jh765              | 1             | [32m](https://app.flowwer.dev/charts/review-time/~(u~(i~'96186164~n~'jh765)~p~30~r~(~(d~'rbd32j~t~'16u)~(d~'rbd3mk~t~'1qv))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        | 2              |

### CODE REVIEW

File: prime-router/metadata/fhir_mapping/hl7/datatype/ContactPoint.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+     condition: $equipmentType EQUALS CP
+     type: STRING
+     valueOf: $mobile
+     expressionType: HL7Spec
+     vars:
+         equipmentType: String, XTN.3
+     constants:
+         mobile: "mobile"
+
+use_3:
```
Comment (carlosfelix2) [inline prime-router/metadata/fhir_mapping/hl7/datatype/ContactPoint.yml:54]: I added use 3 and 4

File: prime-router/metadata/fhir_mapping/hl7/datatype/ContactPoint.yml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    vars:
+       start: XTN.13
+       end: XTN.14
+
+rank: 
+     type: STRING
+     valueOf: XTN.18
+     expressionType: HL7Spec
+
+extension_1:
```
Comment (carlosfelix2) [inline prime-router/metadata/fhir_mapping/hl7/datatype/ContactPoint.yml:80]: I added this text extension

File: prime-router/metadata/fhir_mapping/hl7/datatype/Identifier_SystemID.yml [hunk 3]
```diff
@@ -54,13 +54,12 @@ extension_1:
   generateList: true
   valueOf: extension/Extension
   expressionType: resource
-  condition: $idStr NOT_NULL && $systemStr EQUALS $isoSystemStr
+  condition: $idStr NOT_NULL && $systemStr EQUALS ISO
```
Comment (carlosfelix2) [inline prime-router/metadata/fhir_mapping/hl7/datatype/Identifier_SystemID.yml:57]: This is just cleanup.  I learned something new

File: prime-router/metadata/fhir_mapping/hl7/resource/DiagnosticReport.yml [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+    obx2: STRING, OBX.2
+  constants:
+    system: "http://terminology.hl7.org/CodeSystem/v2-0834"
+    code: "text/plain"
+    display: "Text data"
+    system2: "http://hl7.org/fhir/ValueSet/all-languages"
+    code2: "en"
+    display2: "English"
+
+extension_1:
```
Comment (carlosfelix2) [inline prime-router/metadata/fhir_mapping/hl7/resource/DiagnosticReport.yml:172]: I added this extension

File: prime-router/metadata/fhir_mapping/hl7/resource/Practitioner.yml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    generateList: true
+    expressionType: resource
+
+address:
+  valueOf: datatype/Address
+  expressionType: resource
+  generateList: true
+  specs: $addressSpec
+
+extension_1:
```
Comment (carlosfelix2) [inline prime-router/metadata/fhir_mapping/hl7/resource/Practitioner.yml:32]: I added extension 1 and 2

File: prime-router/metadata/fhir_mapping/hl7/resource/PractitionerRole.yml [hunk 6]
```diff
@@ -0,0 +1,24 @@
+resourceType: PractitionerRole
```
Comment (carlosfelix2) [inline prime-router/metadata/fhir_mapping/hl7/resource/PractitionerRole.yml:1]: This is a 100% custom resource not provided by the library

File: prime-router/metadata/fhir_mapping/hl7/resource/ServiceRequest.yml [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+    codeCWE: OBR.4
+
+note:
+  valueOf: datatype/Annotation
+  condition: $annotationText NOT_NULL
+  expressionType: resource
+  vars:
+    annotationText: NTE.3 *&, GeneralUtils.concatenateWithChar(annotationText, '  \n')
+
+extension_1:
```
Comment (carlosfelix2) [inline prime-router/metadata/fhir_mapping/hl7/resource/ServiceRequest.yml:177]: I added extension 1 and 2

**Review by clediggins-usds — APPROVED** [review]

**Pull request merged**